src/Framework/Blocks/BlockCollection.php

```diff
@@ -79,24 +79,107 @@ public function getBlocks(): array
     /**
      * @unreleased
      *
-     * @param BlockModel[] $blocks
-     * @param string $name
-     * @return BlockModel|null
+     * @return BlockCollection|null
      */
-    public function findByName(string $name, array $blocks = null)
+    public function findBlock(string $blockName, int $blockIndex = 0, BlockCollection $blockCollection = null, string $return = 'self', int &$count = 0)
     {
-        if ($blocks === null) {
-            $blocks = $this->blocks;
+        if ($blockCollection === null) {
+            $blockCollection = $this;
         }
 
-        foreach ($blocks as $block) {
-            if ($block->name === $name) {
-                return $block;
-            } else if ($block->innerBlocks) {
-                return $this->findByName($name, $block->innerBlocks->blocks);
+        foreach ($blockCollection->blocks as $block) {
+            if ($block->name === $blockName) {
+                $count++;
+
+                if ($count === $blockIndex + 1) {
+                    if ($return === 'self') {
+                        return $block->innerBlocks;
+                    } elseif ($return === 'parent') {
+                        return $blockCollection;
+                    }
+                }
+            } elseif ($block->innerBlocks) {
+                $result = $this->findBlock($blockName, $blockIndex, $block->innerBlocks, $return, $count);
+                if ($result) {
+                    return $result;
+                }
             }
         }
 
         return null;
     }
+
+    /**
+     * @unreleased
+     */
+    public function insertBefore(string $blockName, BlockModel $block, int $blockIndex = 0): BlockCollection
+    {
+        $blockCollection = $this->findBlock($blockName, $blockIndex, $this, 'parent');
+
+        if (!$blockCollection) {
+            return $this;
+        }
+
+        $innerBlocks = $blockCollection->blocks;
+        $blockIndex = array_search($blockName, array_column($innerBlocks, wp_is_uuid($blockName) ? 'id' : 'name'));
+        array_splice($innerBlocks, $blockIndex, 0, [$block]);
+        $blockCollection->blocks = $innerBlocks;
+
+        return $this;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function insertAfter(string $blockName, BlockModel $block, int $blockIndex = 0): BlockCollection
+    {
+        $blockCollection = $this->findBlock($blockName, $blockIndex, $this, 'parent');
+
+        if (!$blockCollection) {
+            return $this;
+        }
+
+        $innerBlocks = $blockCollection->blocks;
+        $blockIndex = array_search($blockName, array_column($innerBlocks, wp_is_uuid($blockName) ? 'id' : 'name'));
```

Comment:
Is this leftover from the client id search?

Reply:
Resolved 1289cc03d34e67991132bfa4ba7beba5781ff219